### PR TITLE
OCPCLOUD-2514,OCPCLOUD-2516: Remove reliance on feature gates for cloud provider flags

### PR DESCRIPTION
**Pull request by JoelSpeed**
In 4.15 the final feature gates were promoted for the CCMs, all in-tree code is now disabled and the decision as to whether the `--cloud-provider` flag should be set to external has been simplified to a case of, which platform are we on.
Additionally the `--cloud-config` flag is no longer required at all.

This PR removes the reliance on the feature gate, removes the cloud config flag and then removes a bunch of feature gates being passed around where they aren't needed.
The latter part of this may need to be reverted in the future if templates regain a need for feature gates, but in theory it should be pretty easy to plumb back through by reverting the commit.

### CODE REVIEW

**Review by cdoern — APPROVED** [review]
looks good but: 

> The latter part of this may need to be reverted in the future if templates regain a need for feature gates, but in theory it should be pretty easy to plumb back through by reverting the commit.

concerns me a bit. I know we put a lot of this plumbing around the MCC for the MCN featuregate. Is there any benefit to leaving this stuff in since we are adding some features behind FG in 4.16?

**Comment by JoelSpeed** [comment]
> I know we put a lot of this plumbing around the MCC for the MCN featuregate. Is there any benefit to leaving this stuff in since we are adding some features behind FG in 4.16?

I deliberately put this as a separate commit as I wasn't sure if you'd want this unwound or not. Currently, there's nothing within the template controller that requires feature gates, so the PR cleans up the unused variables. I can however revert this last commit and the code still compiles, let me know which you'd prefer, to keep the plumbing or clean it up.

Are you aware of any of the TP features needing to use the template controller?

File: pkg/controller/container-runtime-config/container_runtime_config_controller.go
Comment (yuqi-zhang) [inline pkg/controller/container-runtime-config/container_runtime_config_controller.go:107]: re: containerruntimeconfigcontroller specifically, I know @QiWang19 has been working on https://github.com/openshift/machine-config-operator/pull/4160 as a FG'ed feature. Would that conflict at all?

File: pkg/controller/template/render.go
Comment (yuqi-zhang) [inline pkg/controller/template/render.go:39]: I don't think the template/render controller controls any FG'ed items and there shouldn't be any plans to immediately, so I'm fine with this (but please correct me if I'm missing something)

**Review by yuqi-zhang — COMMENTED** [review]
I think this should be safe, one question on the containerruntimeconfigcontroller inline

File: pkg/controller/container-runtime-config/container_runtime_config_controller.go
Comment (QiWang19) [reply pkg/controller/container-runtime-config/container_runtime_config_controller.go:107]: yes, it's a conflict, [#4160](https://github.com/openshift/machine-config-operator/pull/4160/files#diff-f0fe3160e03c1ef1c0b9a88475ba02b933160e98f25c648b165b89c93d42c107R360) relies on this field to check if the API that containerruntime config watching for is featuregate enabled.

Comment (JoelSpeed) [reply pkg/controller/container-runtime-config/container_runtime_config_controller.go:107]: How do you want to handle this then? I can drop the specific CRC parts, drop the whole commit, or Qi can handle the conflict if this merges first. Any preference?

Comment (yuqi-zhang) [reply pkg/controller/container-runtime-config/container_runtime_config_controller.go:107]: I don't have strong feelings here, @QiWang19 what would work for you?

Comment (QiWang19) [reply pkg/controller/container-runtime-config/container_runtime_config_controller.go:107]: can we drop the specific CRC parts from this PR? so we don't have to have separate PR deleting and adding the same lines of code.

Comment (JoelSpeed) [reply pkg/controller/container-runtime-config/container_runtime_config_controller.go]: I've reviewed the other PR and updated the unwinding here to not conflict, I've still removed from `generateOriginalContainerRuntimeConfigs` but as far as I can tell that shouldn't conflict with your PR

Comment (QiWang19) [reply pkg/controller/container-runtime-config/container_runtime_config_controller.go]: @JoelSpeed Thanks

**Review by yuqi-zhang — APPROVED** [review]
/lgtm

updated changes look good, thanks!

File: pkg/controller/container-runtime-config/container_runtime_config_bootstrap.go
Comment (yuqi-zhang) [inline pkg/controller/container-runtime-config/container_runtime_config_bootstrap.go:36]: verify is complaining that the function header still has featuregate being passed in

File: pkg/controller/container-runtime-config/container_runtime_config_controller.go
Comment (yuqi-zhang) [inline pkg/controller/container-runtime-config/container_runtime_config_controller.go:906]: same for the `registriesConfigIgnition` function

**Comment by QiWang19** [comment]
For [adbd27b](https://github.com/openshift/machine-config-operator/pull/4228/commits/adbd27b902f64047b7a16c9195dbee589b8caeea), I can handle the conflict when after this merged.

File: pkg/controller/container-runtime-config/container_runtime_config_controller.go
Comment (yuqi-zhang) [inline pkg/controller/container-runtime-config/container_runtime_config_controller.go:897]: Now this is complaining about `Category: unparam` instead

Since Qi is fine with resolving conflicts, maybe it's easier to remove them. I'm fine either way

**Comment by JoelSpeed** [comment]
@yuqi-zhang @QiWang19 I've got the nolint directive working, so Qi if you could unwind the nolint in your PR, that would be appreciated

**Review by damdo — COMMENTED** [review]
Re-adding LGTM after lint fix
/lgtm

**Comment by QiWang19** [comment]
@JoelSpeed  Thanks for your consideration.

**Pull request merged**
